### DISCUSSION TITLE
[FEATURE] Crossfaction ICC Gunship Battle / Vendors

### DISCUSSION THREAD

**Discussion by disclosurez** [azerothcore/azerothcore-wotlk#3923] [Ideas]
**Is your feature request related to a problem? Please describe.**
ICC & Crossfaction do not mix together well at all.
Please can we come up with a solution for this.

**Describe the solution you'd like**
ICC dynamically changes the gunship battle / vendors to the leader of the groups faction, thus not ruining the fight if half the group is horde and the other half alliance.

**Describe alternatives you've considered**
Hackfixes suck, I can set vendors / non boss npcs to different factions but its not very well. 
The gunship bosses have to stay friendly to all during cross faction, and well it breaks the fight.

**Additional context**

**Comment by FrancescoBorzi** [comment]
@Winfidonarleyan

**Comment by Winfidonarleyan** [comment]
Nothing is clear.

**Comment by disclosurez** [comment]
> Nothing is clear.

Sorry, what do you mean by this?

Have you ever tried to do the gunship battle with half of the raid being friendly to it? And the other half hostile?

That is the issue here, we need something to dynamically change the npc's faction to 48 (hostile to all or something)

Thoughts?

**Comment by Winfidonarleyan** [comment]
No thoughts

**Comment by mik1893** [comment]
i did an hackfix on my project ages ago... i'll need to see if i can find it

**Reply by Fireevil** [reply]
any fix for this bug ?

**Comment by Fireevil** [comment]
any fix for this bug ?

**Reply by Froscht** [reply]
i did gunship is alway horde  and set npcs needed to faction 35 etc.
